src/helpers/convertToTypesFromSchemaProperties.ts

```diff
@@ -0,0 +1,89 @@
+import { getResultStringForStringType } from './getResultStringForStringType';
+import { getResultStringType } from './getResultStringType';
+import { parseFormat } from './parseFormat';
+import { parseProperty } from './parseProperty';
+import { parseRefType } from './parseRefType';
+
+export const convertToTypesFromSchemaProperties = ({
+    schema,
+    schemaKey,
+    interfaces,
+}: {
+    schema: SwaggerSchema;
+    schemaKey?: string;
+    interfaces?: Array<string>;
+}): string => {
+    const getStandardString = ({ propertyName, description, nullable, refType, format, isArray }: any) => {
+        return `${parseProperty({ propertyName, description, nullable })}${parseRefType(refType)}${
+            isArray ? '[]' : ''
+        };${parseFormat(format)}\n`;
+    };
+
+    let result = `${getDescription({ description: schema.description })}export interface ${schemaKey}${
+        interfaces ? ` extends ${interfaces.join(', ')} ` : ' '
+    }{\n`;
+
+    if (schema.properties) {
+        getSchemaProperties(schema.properties).map(props => {
+            const resultStringType = getResultStringType(props);
+
+            switch (resultStringType) {
+                case 'String':
+                    result += getResultStringForStringType(props);
+                    break;
+                case 'Number':
+                    result += getResultStringForNumberType(props);
+                    break;
+                case 'Boolean':
+                    result += getResultStringForBooleanType(props);
+                    break;
+                case 'ItemWithRef':
+                    result += getResultStringForArrayType({
+                        ...props,
+                        refType: props.items[SwaggerProps.$ref].split('/'),
+                    });
+                    break;
+                case 'ItemWithoutRef':
+                    result += getResultStringForArrayWithoutItemRef(props);
+                    break;
+                case 'OneOf':
+                    result += getStandardString({
+                        ...props,
+                        refType: props.oneOf[0][SwaggerProps.$ref].split('/'),
+
+                        isArray: false,
+                    });
+                    break;
+                case 'Ref':
+                    result += getStandardString({ ...props, refType: props.$ref.split('/'), isArray: false });
+                    break;
+                case 'UndefinedTypeRefAndOneOf':
+                    result += `    ${props.propertyName}${props.nullable ? '?' : ''}: any;\n`;
+                    break;
+                case 'DictionaryKey':
+                    result += getResultStringForDictionaryKey(props);
+                    break;
+                case 'AdditionalPropertiesType':
+                    result += getResultStringForAdditionalPropertiesType(props);
+                    break;
+                case 'Invalid':
+                default:
```

Comment:
Also added this fallback `invalid` and `default` , we didn't have it before and we fail silently, like in the bug I have to fix herehttps://github.com/LandrAudio/openapi-codegen-typescript/pull/53